cache/lru_cache.cc

```diff
@@ -461,8 +461,10 @@ std::string LRUCacheShard::GetPrintableOptions() const {
 }
 
 LRUCache::LRUCache(size_t capacity, int num_shard_bits,
-                   bool strict_capacity_limit, double high_pri_pool_ratio)
-    : ShardedCache(capacity, num_shard_bits, strict_capacity_limit) {
+                   bool strict_capacity_limit, double high_pri_pool_ratio,
+                   std::shared_ptr<CacheAllocator> cache_allocator)
```

Comment:
nit: Pass lvalue reference instead? That way you don't need to write `std::move` everywhere.

Reply:
Saves me one copy though ;)

Reply:
How does it save one copy? There's still one copy of the shared_ptr when `NewLRUCache` is being called, if pass by value. If pass by reference, there's still one copy, to store the value in `cache.cache_allocator_`.

Reply:
If the client decides to `std::move()` cache_allocator into the `NewLRUCache` call then there's no copy at all. Of course, the client can decide to keep its own reference, in which case the copy is made. But passing the parameter by value lets the client decide that.

Reply:
ok, make sense.